### PR TITLE
HAWKULAR-1327 Add versioning on resource types

### PR DESCRIPTION
**Pull request by lucasponce**

### CODE REVIEW

**Comment by lucasponce** [comment]
image

**Comment by lucasponce** [comment]
I wonder if we miss a WF11 and EAP7 config files, too. But I guess those can be in a following PR when needed.

File: eap6-support/hawkular-javaagent-wildfly-feature-pack-eap6/src/main/resources/featurepack/content/standalone/configuration/hawkular-javaagent-config.yaml [hunk 1]
```diff
@@ -551,10 +552,6 @@ resource-type-set-dmr:
       attribute: product-name
     - name: Version
       attribute: product-version
-    - name: Running Mode
-      attribute: running-mode
-    - name: Server State
-      attribute: server-state
```
Comment (jmazzitelli) [inline eap6-support/hawkular-javaagent-wildfly-feature-pack-eap6/src/main/resources/featurepack/content/standalone/configuration/hawkular-javaagent-config.yaml]: need to put these back in. these do exist and IIRC MiQ uses them. I get this in my EAP 6.4 server:

/:read-attribute(name=server-state)
{
    "outcome" => "success",
    "result" => "running"
}

/:read-attribute(name=running-mode)
{
    "outcome" => "success",
    "result" => "NORMAL"
}

/:read-attribute(name=product-version)
{
    "outcome" => "success",
    "result" => "6.4.0.GA"
}

File: eap6-support/hawkular-javaagent-wildfly-feature-pack-eap6/src/main/resources/featurepack/content/standalone/configuration/hawkular-javaagent-config.yaml [hunk 2]
```diff
@@ -567,6 +564,7 @@ resource-type-set-dmr:
     - name: Bound Address
       attribute: bound-address
       path:     /socket-binding-group=standard-sockets/socket-binding=http
+      # TODO Do we need a https one ?
```
Comment (jmazzitelli) [inline eap6-support/hawkular-javaagent-wildfly-feature-pack-eap6/src/main/resources/featurepack/content/standalone/configuration/hawkular-javaagent-config.yaml]: no, MiQ doesn't need / use it. We'll add it if they want it later.

**Review by jmazzitelli — CHANGES_REQUESTED** [review]

File: eap6-support/hawkular-javaagent-wildfly-feature-pack-eap6/src/main/resources/featurepack/content/standalone/configuration/hawkular-javaagent-config.yaml (hunk 1 above)
Comment (lucasponce) [reply eap6-support/hawkular-javaagent-wildfly-feature-pack-eap6/src/main/resources/featurepack/content/standalone/configuration/hawkular-javaagent-config.yaml]: Ah, I've tested in 6.4.7 and I didn't see them, I will double check.
Yes, if they are in 6.4 it should be back.
Thanks for checking it.

**Pull request merged**